integration/log.go

```diff
@@ -304,70 +303,63 @@ func readbackLogEntries(logID int64, client trillian.TrillianLogClient, params T
 		}
 		return r
 	}(expect)
-
-	currentLeaf := int64(0)
-	leafMap := make(map[int64]*trillian.LogLeaf)
 	glog.Infof("Expecting %d unique leaves", len(expect))
 
-	for currentLeaf < params.LeafCount {
-		// We have to allow for the last batch potentially being a short one
-		numLeaves := params.LeafCount - currentLeaf
-
-		if numLeaves > params.ReadBatchSize {
-			numLeaves = params.ReadBatchSize
+	var leaves []*trillian.LogLeaf
```

Comment:
It would be reasonable to pre-allocate this array based on the `params.LeafCount`.